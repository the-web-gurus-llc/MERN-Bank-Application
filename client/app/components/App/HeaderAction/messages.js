import { defineMessages } from 'react-intl';

export const scope = 'app.components.HeaderAction';

export default defineMessages({
  messages: {
    id: `${scope}.messages`,
    defaultMessage: 'Messages',
  },
  notifications: {
    id: `${scope}.notifications`,
    defaultMessage: 'Notifications',
  },
  logout: {
    id: `${scope}.logout`,
    defaultMessage: 'Logout',
  },
  popConfirmTitle: {
    id: `${scope}.pop.confirm.title`,
    defaultMessage: 'Are you sure you want to log out?',
  },
  popConfirmOk: {
    id: `${scope}.pop.confirm.ok`,
    defaultMessage: 'Yes',
  },
  popConfirmCancel: {
    id: `${scope}.pop.confirm.cancel`,
    defaultMessage: 'Cancel',
  },
});
