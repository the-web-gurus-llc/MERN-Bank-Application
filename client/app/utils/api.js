const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
