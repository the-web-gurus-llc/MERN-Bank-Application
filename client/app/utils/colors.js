export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
