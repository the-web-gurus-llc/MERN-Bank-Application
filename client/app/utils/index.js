export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
