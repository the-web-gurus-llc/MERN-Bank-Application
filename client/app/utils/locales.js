exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
