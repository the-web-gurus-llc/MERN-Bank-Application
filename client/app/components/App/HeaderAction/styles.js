import styled from 'styled-components';
import { media, colors } from 'utils';
import {
  PoweroffOutlined,
  MessageOutlined,
  BellOutlined,
  BellFilled,
  MessageFilled,
} from '@ant-design/icons';
import { headerHeight } from 'components/App/Header/styles';
import { Button } from 'antd';

export const StyledHeaderAction = styled.div`
  color: black;
  width: 100%;
  display: flex;
  justify-content: space-evenly;
  height: ${headerHeight};
  align-items: center;

  ${media.tablet`
    width: auto;
    margin-right: 55px;
  `};
`;

export const StyledHeaderActionItem = styled(Button)`
  display: inline-flex;
  align-items: center;
  padding: 0;

  .anticon {
    display: flex;
  }
`;

export const StyledHeaderActionItemName = styled.div`
  display: none;
  color: ${colors.black};

  ${media.tablet`
    margin-left: 10px;
    display: inline-block;
  `}
`;

export const StyledPoweroffOutlined = styled(PoweroffOutlined)`
  font-size: 19px;
`;

export const StyledMessageOutlined = styled(MessageOutlined)`
  font-size: 19px;
`;

export const StyledMessageFilled = styled(MessageFilled)`
  font-size: 19px;
`;

export const StyledBellOutlined = styled(BellOutlined)`
  font-size: 19px;
`;

export const StyledBellFilled = styled(BellFilled)`
  font-size: 19px;
`;

export const StyledHeaderWrapper = styled.div`
  display: flex;
  border-radius: 5px;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  justify-content: space-around;
  width: 90%;

  ${media.tablet`
    width: 100%;

    button {
      padding: 0 10px;

      &:hover {
        background: rgba(0, 0, 0, 0.08);
      }
    }
  `}
`;
