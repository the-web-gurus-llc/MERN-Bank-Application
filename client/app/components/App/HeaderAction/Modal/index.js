import React from 'react';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { useDispatch, useSelector } from 'react-redux';
import { makeSelectIsOpenedModal } from 'containers/App/selectors';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { getRequestName } from 'helpers';
import { LOGOUT_REQUEST } from 'containers/App/constants';
import { logoutAction, toggleConfirmModalAction } from 'containers/App/actions';
import { StyledModal } from 'components/App/Modal/styles';
import { Button } from 'antd';
import LoadingIndicator from 'components/LoadingIndicator';
import messages from '../messages';

const stateSelector = createStructuredSelector({
  isOpenedModal: makeSelectIsOpenedModal(),
  isLoading: makeSelectIsLoading(getRequestName(LOGOUT_REQUEST)),
});

export default function Modal() {
  const { isOpenedModal, isLoading } = useSelector(stateSelector);
  const dispatch = useDispatch();

  const onLogout = () => dispatch(logoutAction());
  const onToggleConfirmModal = () => dispatch(toggleConfirmModalAction());

  return (
    <FormattedMessage {...messages.logout}>
      {(title) => (
        <StyledModal
          title={title}
          visible={isOpenedModal}
          onOk={onLogout}
          onCancel={onToggleConfirmModal}
          footer={[
            <Button key="back" onClick={onToggleConfirmModal}>
              <FormattedMessage {...messages.popConfirmCancel} />
            </Button>,
            <Button
              key="submit"
              disabled={isLoading}
              type="primary"
              onClick={onLogout}
            >
              {isLoading ? (
                <LoadingIndicator />
              ) : (
                <FormattedMessage {...messages.popConfirmOk} />
              )}
            </Button>,
          ]}
        >
          <FormattedMessage {...messages.popConfirmTitle} />
        </StyledModal>
      )}
    </FormattedMessage>
  );
}
