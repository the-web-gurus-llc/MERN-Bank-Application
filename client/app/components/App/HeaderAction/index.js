import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  logoutAction,
  getMessagesAction,
  getNotificationsAction,
  toggleConfirmModalAction,
} from 'containers/App/actions';
import { Popconfirm, Badge, Dropdown } from 'antd';
import { FormattedMessage, intlShape, injectIntl } from 'react-intl';
import { useMediaQuery } from 'react-responsive';
import { createStructuredSelector } from 'reselect';
import { makeSelectUser, makeSelectMessages } from 'containers/App/selectors';
import {
  StyledHeaderAction,
  StyledHeaderActionItem,
  StyledHeaderActionItemName,
  StyledMessageOutlined,
  StyledBellOutlined,
  StyledPoweroffOutlined,
  StyledHeaderWrapper,
  StyledBellFilled,
  StyledMessageFilled,
} from './styles';
import messages from './messages';
import Messages from '../Messages';
import Notifications from '../Notifications';
import Modal from './Modal';

const stateSelector = createStructuredSelector({
  user: makeSelectUser(),
  messagesData: makeSelectMessages(),
});

function HeaderAction({ intl }) {
  const { messagesData, user } = useSelector(stateSelector);
  const dispatch = useDispatch();
  const isMobile = useMediaQuery({ maxWidth: 479 });

  const onLogout = () => dispatch(logoutAction());
  const onGetMessages = () => dispatch(getMessagesAction());
  const onGetNotifications = () => dispatch(getNotificationsAction());
  const onToggleConfirmModal = () => dispatch(toggleConfirmModalAction());

  return (
    <StyledHeaderAction>
      <StyledHeaderWrapper>
        <Dropdown
          trigger={['click']}
          overlay={<Messages />}
          placement="bottomCenter"
          arrow={!isMobile}
          getPopupContainer={(trigger) => trigger.parentNode}
        >
          <StyledHeaderActionItem
            type="link"
            onClick={!messagesData?.data?.length && onGetMessages}
          >
            <Badge count={user?.userConfig?.messageCount}>
              {user?.userConfig?.messageCount ? (
                <StyledMessageFilled />
              ) : (
                <StyledMessageOutlined />
              )}
            </Badge>
            <StyledHeaderActionItemName>
              <FormattedMessage {...messages.messages} />
            </StyledHeaderActionItemName>
          </StyledHeaderActionItem>
        </Dropdown>

        <Dropdown
          trigger={['click']}
          overlay={<Notifications />}
          placement="bottomCenter"
          arrow={!isMobile}
          getPopupContainer={(trigger) => trigger.parentNode}
        >
          <StyledHeaderActionItem
            type="link"
            onClick={user?.userConfig?.notificationCount && onGetNotifications}
          >
            <Badge count={user?.userConfig?.notificationCount}>
              {user?.userConfig?.notificationCount ? (
                <StyledBellFilled />
              ) : (
                <StyledBellOutlined />
              )}
            </Badge>
            <StyledHeaderActionItemName>
              <FormattedMessage {...messages.notifications} />
            </StyledHeaderActionItemName>
          </StyledHeaderActionItem>
        </Dropdown>

        {isMobile ? (
          <>
            <StyledHeaderActionItem type="link" onClick={onToggleConfirmModal}>
              <StyledPoweroffOutlined />
              <StyledHeaderActionItemName>
                <FormattedMessage {...messages.logout} />
              </StyledHeaderActionItemName>
            </StyledHeaderActionItem>

            <Modal />
          </>
        ) : (
          <Popconfirm
            placement="bottomRight"
            title={intl.formatMessage(messages.popConfirmTitle)}
            onConfirm={onLogout}
            okText={intl.formatMessage(messages.popConfirmOk)}
            cancelText={intl.formatMessage(messages.popConfirmCancel)}
          >
            <StyledHeaderActionItem type="link">
              <StyledPoweroffOutlined />
              <StyledHeaderActionItemName>
                <FormattedMessage {...messages.logout} />
              </StyledHeaderActionItemName>
            </StyledHeaderActionItem>
          </Popconfirm>
        )}
      </StyledHeaderWrapper>
    </StyledHeaderAction>
  );
}

HeaderAction.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(HeaderAction);
