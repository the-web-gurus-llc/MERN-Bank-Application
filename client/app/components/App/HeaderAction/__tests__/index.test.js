import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'utils/configureStore';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import HeaderAction from '../index';

describe('<HeaderAction />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a HeaderAction', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <HeaderAction />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
