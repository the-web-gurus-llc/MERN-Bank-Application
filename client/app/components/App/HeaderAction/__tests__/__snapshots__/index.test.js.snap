// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderAction /> should render a HeaderAction 1`] = `
<div
  class="HeaderActionstyle__StyledHeaderAction-sc-1v0240u-0 feaNdT"
>
  <div
    class="HeaderActionstyle__StyledHeaderWrapper-sc-1v0240u-8 hQPoLy"
  >
    <button
      class="ant-btn HeaderActionstyle__StyledHeaderActionItem-sc-1v0240u-1 cmPPXO ant-dropdown-trigger ant-btn-link"
      type="button"
    >
      <span
        class="ant-badge"
      >
        <span
          aria-label="message"
          class="anticon anticon-message HeaderActionstyle__StyledMessageOutlined-sc-1v0240u-4 iIaPJm"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="message"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
            />
          </svg>
        </span>
      </span>
      <div
        class="HeaderActionstyle__StyledHeaderActionItemName-sc-1v0240u-2 hgmiCZ"
      >
        <span>
          Messages
        </span>
      </div>
    </button>
    <button
      class="ant-btn HeaderActionstyle__StyledHeaderActionItem-sc-1v0240u-1 cmPPXO ant-dropdown-trigger ant-btn-link"
      type="button"
    >
      <span
        class="ant-badge"
      >
        <span
          aria-label="bell"
          class="anticon anticon-bell HeaderActionstyle__StyledBellOutlined-sc-1v0240u-6 fDtdaM"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="bell"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
            />
          </svg>
        </span>
      </span>
      <div
        class="HeaderActionstyle__StyledHeaderActionItemName-sc-1v0240u-2 hgmiCZ"
      >
        <span>
          Notifications
        </span>
      </div>
    </button>
    <button
      class="ant-btn HeaderActionstyle__StyledHeaderActionItem-sc-1v0240u-1 cmPPXO ant-btn-link"
      type="button"
    >
      <span
        aria-label="poweroff"
        class="anticon anticon-poweroff HeaderActionstyle__StyledPoweroffOutlined-sc-1v0240u-3 dauYlO"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="poweroff"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M705.6 124.9a8 8 0 00-11.6 7.2v64.2c0 5.5 2.9 10.6 7.5 13.6a352.2 352.2 0 0162.2 49.8c32.7 32.8 58.4 70.9 76.3 113.3a355 355 0 0127.9 138.7c0 48.1-9.4 94.8-27.9 138.7a355.92 355.92 0 01-76.3 113.3 353.06 353.06 0 01-113.2 76.4c-43.8 18.6-90.5 28-138.5 28s-94.7-9.4-138.5-28a353.06 353.06 0 01-113.2-76.4A355.92 355.92 0 01184 650.4a355 355 0 01-27.9-138.7c0-48.1 9.4-94.8 27.9-138.7 17.9-42.4 43.6-80.5 76.3-113.3 19-19 39.8-35.6 62.2-49.8 4.7-2.9 7.5-8.1 7.5-13.6V132c0-6-6.3-9.8-11.6-7.2C178.5 195.2 82 339.3 80 506.3 77.2 745.1 272.5 943.5 511.2 944c239 .5 432.8-193.3 432.8-432.4 0-169.2-97-315.7-238.4-386.7zM480 560h64c4.4 0 8-3.6 8-8V88c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8z"
          />
        </svg>
      </span>
      <div
        class="HeaderActionstyle__StyledHeaderActionItemName-sc-1v0240u-2 hgmiCZ"
      >
        <span>
          Logout
        </span>
      </div>
    </button>
  </div>
</div>
`;
